compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -259,11 +259,11 @@ fn check_item<'tcx>(tcx: TyCtxt<'tcx>, item: &'tcx hir::Item<'tcx>) -> Result<()
                     .emit());
             }
             // We match on both `ty::ImplPolarity` and `ast::ImplPolarity` just to get the `!` span.
-            match tcx.impl_polarity(def_id) {
-                ty::ImplPolarity::Positive => {
+            match header.map(|h| h.skip_binder().polarity) {
+                Some(ty::ImplPolarity::Positive) | None => {
```

Comment:
A comment here would be useful that `None` means `impl Inherent {}`, which can never be unsafe.